index_copy.js

```diff
@@ -0,0 +1,104 @@
+const battleship = () => {
+
+    // create objects with the given parametrs
+    let playerOne = {
+      name: "",
+      shipCount: 0,
+      gameBoard: []
+    }
+    let playerTwo = {
+      name: "",
+      shipCount: 0,
+      gameBoard: []
+    };
+  
+    // asking for the players' name
+    playerOne.name = prompt("Player 1 what is your name?");
+    playerTwo.name = prompt("Player 2 what is your name?");
+  
+    // creating game boards
+    let boardColumn = 4;
+    let shipCountFinal = 4;
+  
+    // add game board with 0
+    for (let i = 0; i < boardColumn; i++) {
+      let boardRow = []
+      for (let j = 0; j < boardColumn; j++) {
+        boardRow.push(0)
+      }
+  
+  
+  // add rows to the end of an empty list
+      playerOne.gameBoard.push(boardRow);
+      playerTwo.gameBoard.push(boardRow);
+    }
```

Comment:
try adding in this line of code here to see how, even before you start the shipPlace work, if you edit one of the arrays, it actually edits both of them.
```
playerTwo.gameBoard[0][0] = 1;
console.log("Test", playerTwo.gameBoard, playerOne.gameBoard);
```